presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/JdbcModule.java

```diff
@@ -71,4 +77,22 @@ public static ConnectionFactory createConnectionFactoryWithStats(ConnectionFacto
     {
         return new StatisticsAwareConnectionFactory(connectionFactory);
     }
+
+    public static class ConnectorFactoryCleanup
+    {
+        private final ConnectionFactory connectionFactory;
+
+        @Inject
+        public ConnectorFactoryCleanup(ConnectionFactory connectionFactory)
+        {
+            this.connectionFactory = requireNonNull(connectionFactory, "connectionFactory is null");
+        }
+
+        @PreDestroy
+        public void shutdown()
+                throws SQLException
+        {
+            connectionFactory.close();
```

Comment:
This class has no other purpose than calling close on `ConnectionFactory`
Since `ConnectionFactory` is managed by guice, why can't it take responsibility for its shutdown, just like all other guice-managed classes do?

Reply:
I just did that in the first place.